tests/Features/ReturnTypes/TestCaseExtension.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Features\ReturnTypes;
+
+use App\User;
+use Illuminate\Foundation\Application;
+use PHPStan\Testing\TestCase;
+
+class TestCaseExtension extends TestCase
+{
+    public function testTestCaseExtension(): void
+    {
+        $testCase = new TestTestCase();
+
+        $testCase->testMockingMethod('mock');
+        $testCase->testMockingMethod('partialMock');
+        $testCase->testMockingMethod('spy');
+    }
+}
+
+class TestTestCase extends \Illuminate\Foundation\Testing\TestCase
+{
+    public function testMockingMethod($method): void
+    {
+        if (method_exists($this, $method)) {
+            $mock = $this->{$method}(User::class);
```

Comment:
You are calling the methods here dynamically. So when we test statically we can't determine if the extension is really working or not. If I remove the extension from `extension.neon` tests are still passing.

Reply:
You're absolutely right, I had to change it because tests were failing on old Laravel version. (no partialMock method). I will create another PR that will test if disabling extensions results in an error so that is covered as well.